Client/Frontend/Settings/SettingsContentViewController.swift

```diff
@@ -0,0 +1,219 @@
+            error.text = NSLocalizedString("Could not connect to \(settingsTitle.string).", comment: "Error message that is shown in settings when there was a problem loading \(settingsTitle.string).")
+            error.textColor = UIColor.redColor() // Firefox Orange!
+            error.textAlignment = NSTextAlignment.Center
+        }
+        error.hidden = true
+        view.addSubview(error)
+
+        spinner.snp_makeConstraints { make in
+            make.center.equalTo(view)
+            return
+        }
+
+        error.snp_makeConstraints { make in
+            make.center.equalTo(view)
+            make.left.equalTo(view.snp_left).offset(20)
+            make.right.equalTo(view.snp_right).offset(-20)
+            make.height.equalTo(44)
+            return
+        }
+
+        return (view, spinner, error)
+    }
+
+    // The content is ready to be shown.
+    private func onLoaded() {
+        self.timer?.invalidate()
+        self.timer = nil
+        self.isLoaded = true
+    }
+
+    func SELdidTimeOut() {
+        self.timer = nil
+        self.isError = true
+    }
+
+    func webView(webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: NSError) {
+        SELdidTimeOut()
+    }
+
+    func webView(webView: WKWebView, didFailNavigation navigation: WKNavigation!, withError error: NSError) {
+        SELdidTimeOut()
+    }
+
+    func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
+        onLoaded()
+    }
+}
+
+extension SettingsContentViewController: KeyboardHelperDelegate {
+    private func layoutInnerViews() {
+        let keyboardHeight = KeyboardHelper.defaultHelper.currentState?.intersectionHeightForView(self.webView) ?? 0
+
+        self.webView.snp_remakeConstraints { make in
+            make.left.right.top.equalTo(self.view)
+            make.bottom.equalTo(self.view).offset(-keyboardHeight)
+        }
+        if !isLoaded {
+            // If the page has loaded, the interstitial view will have been removed.
+            self.interstitialView.snp_remakeConstraints { make in
+                make.left.right.top.equalTo(self.view)
+                make.bottom.equalTo(self.view).offset(-keyboardHeight)
+            }
+        }
+    }
+
+    func keyboardHelper(keyboardHelper: KeyboardHelper, keyboardWillShowWithState state: KeyboardState) {
+        animateInnerViewsWithKeyboard(state)
+    }
+
+    func keyboardHelper(keyboardHelper: KeyboardHelper, keyboardDidShowWithState state: KeyboardState) {
+        animateInnerViewsWithKeyboard(state)
+    }
+
+    func keyboardHelper(keyboardHelper: KeyboardHelper, keyboardWillHideWithState state: KeyboardState) {
+        animateInnerViewsWithKeyboard(state)
+    }
+
+    private func animateInnerViewsWithKeyboard(keyboardState: KeyboardState) {
+        layoutInnerViews()
+
+        UIView.animateWithDuration(keyboardState.animationDuration, animations: {
```

Comment:
Nit: You can set the animation curve in the options parameters in the extended API for animateWithDuration.